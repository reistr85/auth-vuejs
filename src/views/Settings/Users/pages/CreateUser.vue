<template>
  <div>
    <PageHeader :schema="$schemas.user" />
    <PageContent>
      <DynamicFormPage :schema="$schemas.user" :service="$api.users" />
    </PageContent>
  </div>
</template>

<script>
import PageHeader from '@/components/PageHeader';
import PageContent from '@/components/PageContent';
import DynamicFormPage from '@/components/dynamics/DynamicFormPage';

export default {
  name: 'CreateUser',
  components: { PageHeader, PageContent, DynamicFormPage },
  data() {
    return {};
  }
};
</script>
