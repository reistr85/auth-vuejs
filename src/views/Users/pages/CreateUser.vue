<template>
  <div>
    <PageHeader :schema="schema" />
    <PageContent>
      <DynamicFormPage :schema="schema" />
    </PageContent>
  </div>
</template>

<script>
import PageHeader from '@/components/PageHeader';
import PageContent from '@/components/PageContent';
import UserSchema from '../schemas/UserSchema';
import DynamicFormPage from '@/components/dynamics/DynamicFormPage';

export default {
  name: 'CreateUser',
  components: { PageHeader, PageContent, DynamicFormPage },
  props: {},
  data() {
    return {
      schema: UserSchema,
    }
  },
  methods: {}
}
</script>

<style lang="scss" scoped>

</style>