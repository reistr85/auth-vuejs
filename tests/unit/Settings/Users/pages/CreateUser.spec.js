import { shallowMount } from '@vue/test-utils';
import CreateUser from '@/views/Settings/Users/pages/CreateUser';

describe('Tests component CreateUser', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(CreateUser);
  });

  it('should contais components PageHeader, PageContent and DynamicFormPage', () => {
    expect(wrapper.findComponent({ name: 'PageHeader' }).exists()).toBeTruthy();
    expect(wrapper.findComponent({ name: 'PageContent' }).exists()).toBeTruthy();
    expect(wrapper.findComponent({ name: 'DynamicFormPage' }).exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  })

  it('should exists state schema and service', () => {
    expect(wrapper.vm.schema).toBeTruthy();
    expect(wrapper.vm.service).toBeTruthy();
  })
})