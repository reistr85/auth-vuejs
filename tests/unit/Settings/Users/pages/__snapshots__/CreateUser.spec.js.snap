// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests component CreateUser should contais components PageHeader, PageContent and DynamicFormPage 1`] = `
<div>
  <pageheader-stub schema="[object Object]"></pageheader-stub>
  <pagecontent-stub>
    <dynamicformpage-stub schema="[object Object]" service="[object Object]"></dynamicformpage-stub>
  </pagecontent-stub>
</div>
`;
